java/daikon/dcomp/BuildJDK.java

```diff
@@ -90,288 +57,380 @@ protected boolean exists(String libRt) {
-    JarFile jfile;
-    final String p = potentialJarFileName;
-    try {
-      String jar_name = findRtJarFilename(p);
-      System.out.printf("using jar file %s%n", jar_name);
-      jfile = new JarFile(jar_name);
-    } catch (ZipException e) {
-      throw new ZipException(e.getMessage() + "; filename was " + p);
+    File jrt = new File(JAVA_HOME);
+    if (!jrt.exists()) {
+      System.out.printf("Directory at JAVA_HOME (%s) does not exist.%n", jrt);
+      System.exit(1);
     }
-    return jfile;
-  }
 
-  private static String findRtJarFilename(final String arg) {
-    return new RtJarFinder(arg).findRtJar("/lib/rt.jar /jre/lib/rt.jar");
-  }
+    try {
+      jrt = jrt.getCanonicalFile();
+    } catch (Exception e) {
+      System.out.printf("Error geting canonical file for %s: %s", jrt, e.getMessage());
+      System.exit(1);
+    }
 
-  void copyStreams(InputStream fis, OutputStream fos) throws Exception {
-    byte[] buf = new byte[1024];
-    int i = 0;
-    while ((i = fis.read(buf)) != -1) {
-      fos.write(buf, 0, i);
+    JAVA_HOME = jrt.getAbsolutePath();
+    if (!java_home.startsWith(JAVA_HOME)) {
+      System.out.printf(
+          "JAVA_HOME (%s) does not agree with java.home (%s).%n", JAVA_HOME, java_home);
+      System.out.printf("Please correct your Java environment.%n");
+      System.exit(1);
     }
-    fis.close();
-    fos.close();
   }
 
-  void translate_classes(JarFile jfile, String dest, String prefix, String prefixOfFilesToInclude)
-      throws java.io.IOException {
-
-    // Map from classname to class so we can find out information about
-    // classes we have not yet instrumented.
-    Map<String, JavaClass> classmap = new LinkedHashMap<>();
+  void translate_jar() throws java.io.IOException {
 
+    String jar_name = java_home + "/lib/rt.jar";
+    System.out.printf("using jar file %s%n", jar_name);
     try {
-
-      // Create the destination directory
-      File dfile = new File(dest);
-      dfile.mkdirs();
-
+      JarFile jfile = new JarFile(jar_name);
       // Get each class to be instrumented and store it away
       Enumeration<JarEntry> entries = jfile.entries();
       while (entries.hasMoreElements()) {
         JarEntry entry = entries.nextElement();
         // System.out.printf("processing entry %s%n", entry);
         final String entryName = entry.getName();
-        if (!entryName.startsWith(prefixOfFilesToInclude) && !entryName.startsWith("META-INF")) {
+        if ((entryName.endsWith("/")) || (entryName.endsWith("~"))) {
           continue;
         }
-        if ((entryName.endsWith("/")) || (entryName.endsWith("~"))) {
+
+        // Get the InputStream for this file
+        InputStream is = jfile.getInputStream(entry);
+        class_stream_map.put(entryName, is);
+      }
+    } catch (Exception e) {
+      throw new Error("Problem while reading " + jar_name, e);
+    }
+  }
+
+  void translate_modules() throws java.io.IOException {
```

Comment:
What does this method do?  It's not documented.  Every method should be documented.
The name is rather confusing because this doesn't seem to do any translation, it just collects information.